Brought warning to higher priority than note, etc.

Meshugah opened the pull request:
#5857 

Update functions.rst

Update for warning over note priority

Updated units-and-global-variables.rst

Updated file for warning over note priority

Updated control-structures.rst

Updated priority of warning over note

Updated assembly.rst

Updated priority of warning over code

Updated introduction-to-smart-contracts.rst

Updated priority of warning over note

Update installing-solidity.rst

Corrected control-structures.rst

Corrected white space error

Corrected whitespace

Corrected whitespace introduction to smart contract

<!--### Your checklist for this pull request

Please review the [guidelines for contributing](http://solidity.readthedocs.io/en/latest/contributing.html) to this repository.

Please also note that this project is released with a [Contributor Code of Conduct](CONDUCT.md). By participating in this project you agree to abide by its terms.
-->

### Description

<!--
Please explain the changes you made here.

Thank you for your help!
-->

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

Meshugah committed:
Brought warning to high priority

Update functions.rst

Update for warning over note priority

Updated units-and-global-variables.rst

Updated file for warning over note priority

Updated control-structures.rst

Updated priority of warning over note

Updated assembly.rst

Updated priority of warning over code

Updated introduction-to-smart-contracts.rst

Updated priority of warning over note

Update installing-solidity.rst

Corrected control-structures.rst

Corrected white space error

Corrected whitespace

Corrected whitespace introduction to smart contract

Meshugah commented:
@ChrisChinchilla Let me know if I can improve on this further somehow 😄

bshastry commented on docs/assembly.rst:
```diff
@@ -394,6 +394,16 @@ use ``x_slot``, and to retrieve the byte-offset you use ``x_offset``.
+    To be safe, always clear the data properly before you use it
+    in a context where this is important:
+    ``uint32 x = f(); assembly { x := and(x, 0xffffffff) /* now use x */ }``
+    To clean signed types, you can use the ``signextend`` opcode.
```
Would it be possible to provide an example for the use of `signextend` to clear higher order bits?

Meshugah replied:
I've added an example. Is this correct and would it be what you've requested?

Meshugah committed:
Added example for signextend

Added sample assembly code for signextend

bshastry commented on docs/assembly.rst:
```diff
@@ -394,6 +394,17 @@ use ``x_slot``, and to retrieve the byte-offset you use ``x_offset``.
+    in a context where this is important:
+    ``uint32 x = f(); assembly { x := and(x, 0xffffffff) /* now use x */ }``
+    To clean signed types, you can use the ``signextend`` opcode:
+    ``assembly { signextend(0, x) }``
```
I'm not sure if this is right: 

```
signextend(i, x): sign extend from (i*8+7)th bit counting from least significant
```

so signextend(0,x) will sign extend from the 7th bit from LSB. What happens if you have a signed 32 bit integer (int32). Wouldn't it change the value itself?

Perhaps, this is what you meant?

```
assembly { signextend(<bit_width_of_x>, x) }
```

ChrisChinchilla replied:
@Meshugah Purely for readability, I think this warning block is better under the code block.

ChrisChinchilla commented:
@bshastry This was primarily a formatting PR, if there were errors in the original text, changes might be better in another PR, but I'll check through anyway.

ChrisChinchilla reviewed (changes requested):
From my perspective, the one change I requested aside, this is just what I was looking for @Meshugah

Meshugah committed:
Added Content and Edited for readability

The example for signextend requested by @bshastry was added. Codeblock was moved before warning for readability.

Meshugah committed:
Merge pull request #1 from ethereum/develop

Update

ChrisChinchilla reviewed (approved):
Looks good to me! Thanks for your work @Meshugah

chriseth commented on docs/control-structures.rst:
```diff
@@ -66,6 +66,28 @@ as the actual contract has not been created yet.
 all function arguments have to be copied to memory.
 
+.. warning::
+  Be careful that ``feed.info.value(10).gas(800)`` only locally sets the ``value`` and amount of ``gas`` sent with the function call, and the parentheses at the end perform the actual call. So in this case, the function is not called and the ``value`` and ``gas`` settings are lost.
```
Please undo the move. `.value` is only introduced later on.

ChrisChinchilla replied:
Looks like this was fixed by https://github.com/ethereum/solidity/pull/6846 as in `develop` it's in the right place.

chriseth commented on docs/control-structures.rst:
```diff
@@ -385,6 +385,9 @@ There are two other ways to trigger exceptions: The ``revert`` function can be u
 revert the current call. It is possible to provide a string message containing details about the error
 that will be passed back to the caller.
 
+.. warning::
```
Please undo this move, it refers to the paragraph after the note.

ChrisChinchilla replied:
Looks like this was fixed by #6846 as in develop it's in the right place now.

chriseth commented on docs/installing-solidity.rst:
```diff
@@ -298,16 +298,16 @@ You might want to install ccache to speed up repeated builds.
 CMake will pick it up automatically.
 Building Solidity is quite similar on Linux, macOS and other Unices:
 
+.. warning::
```
This should be moved further down, it distracts the reading flow.

chriseth commented on docs/units-and-global-variables.rst:
```diff
@@ -163,6 +163,16 @@ Mathematical and Cryptographic Functions
 ``keccak256(bytes memory) returns (bytes32)``:
     compute the Keccak-256 hash of the input
 
+.. warning::
```
Please undo this move, it refers to ecrecover, which is only introduced further down.
